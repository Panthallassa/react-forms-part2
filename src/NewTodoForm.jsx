import React, { useState } from "react";

const NewTodoForm = ({ addTodo }) => {
	const [task, setTask] = useState("");

	const handleSubmit = (e) => {
		e.preventDefault();
		if (task) {
			addTodo(task);
			setTask("");
		}
	};

	return (
		<form onSubmit={handleSubmit}>
			<input
				type='text'
				value={task}
				onChange={(e) => setTask(e.target.value)}
				placeholder='Enter new todo'
			/>
			<button type='submit'>Add Todo</button>
		</form>
	);
};

export default NewTodoForm;
