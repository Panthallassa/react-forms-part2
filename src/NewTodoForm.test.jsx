import React from "react";
import { render } from "@testing-library/react";
import NewTodoForm from "./NewTodoForm";

test("renders NewTodoForm", () => {
	const { getByPlaceholderText } = render(
		<NewTodoForm addTodo={() => {}} />
	);
	expect(
		getByPlaceholderText(/Enter new todo/i)
	).toBeInTheDocument();
});

test("matches snapshot", () => {
	const { asFragment } = render(
		<NewTodoForm addTodo={() => {}} />
	);
	expect(asFragment()).toMatchSnapshot();
});
