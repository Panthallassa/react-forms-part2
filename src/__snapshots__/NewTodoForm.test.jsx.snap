// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <input
      placeholder="Enter new todo"
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Add Todo
    </button>
  </form>
</DocumentFragment>
`;
